content/advent-2016/exposing-go-on-the-internet.md

```diff
@@ -0,0 +1,208 @@
++++
+date = "2016-12-19T01:30:27Z"
+author = ["Filippo Valsorda"]
+title = "So you want to expose Go on the Internet"
+linktitle = "Exposing Go on the Internet"
+series = ["Advent 2016"]
++++
+
+Back when `crypto/tls` was slow and `net/http` young, the general wisdom was to always put Go servers behind a reverse proxy like Nginx. That's not necessary anymore!
+
+However, the Internet is the deep end of the pool when it comes to networks, and there are a few things you have to do to teach your server to swim before throwing it in.
+
+At [Cloudflare][cf] we recently experimented with exposing pure Go services to the hostile wide area network. Here are some hard-learned lessons.
+
+[cf]: https://blog.cloudflare.com/tag/go/
+
+## `crypto/tls`
+
+You're not running an insecure HTTP server on the Internet in 2016. So you need `crypto/tls`. The good news is that it's [now really fast][gap] (as you've seen in a [previous article on this blog][bench]), and its security track record so far is excellent.
+
+The default settings resemble the *Intermediate* recommended configuration of the [Mozilla guidelines][mozilla]. However, you should still set `PreferServerCipherSuites` to ensure safer and faster ciphersuites are preferred, and `CurvePreferences` to avoid unoptimized curves: a client using `CurveP384` would cause up to a second of CPU to be consumed on our machines.
+
+```go
+&tls.Config{
+	// Causes servers to use Go's default ciphersuite preferences,
+	// which are tuned to avoid attacks. Does nothing on clients.
+	PreferServerCipherSuites: true,
+	// Only use curves which have assembly implementations
+	CurvePreferences: []tls.CurveID{
+		tls.CurveP256,
+                tls.X25519, // Go 1.8 only
+	},
+}
+```
+
+IF you can take the compatibility loss of the [*Modern* profile][mozilla], you should then also set `MinVersion` and `CipherSuites`.
```

Comment:
I presume you're stressing the `if` here and this isn't a typo? As it kind of looks like one as you've used italics elsewhere.